### PR TITLE
Expose streams in public round APIs

### PR DESCRIPTION
**Pull request by Matt711**
## Description

Contributes to #13744 

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: cpp/tests/streams/round_test.cpp [hunk 1]
```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright (c) 2023-2024, NVIDIA CORPORATION.
+ *
```
Comment (Matt711) [inline cpp/tests/streams/round_test.cpp:3]: TODO: Check if there's already a file where `round` is tested

File: cpp/include/cudf/round.hpp [hunk 2]
```diff
@@ -66,6 +67,7 @@ enum class rounding_method : int32_t { HALF_UP, HALF_EVEN };
  * @param decimal_places Number of decimal places to round to (default 0). If negative, this
  * specifies the number of positions to the left of the decimal point.
  * @param method         Rounding method
+ * @param stream CUDA stream used for device memory operations and kernel launches
```
Comment (ttnghia) [inline cpp/include/cudf/round.hpp:70]: Out of format. Please reformat it.

Comment (Matt711) [reply cpp/include/cudf/round.hpp]: @ttnghia Could you clarify your review?

**Review by ttnghia — APPROVED** [review]

**Review by bdice — APPROVED** [review]

**Pull request merged**